### PR TITLE
Natick mall

### PR DESCRIPTION
**Pull request by adpjay**
This is the scraper for the Natick Mall Mass Vaccination site.

### CODE REVIEW

**Review by livgust — APPROVED** [review]
Thanks, this looks great! Confirmed that it shows no availability, consistent with the UI of the website. Womp womp.

File: site-scrapers/NatickMall.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        ...sites.NatickMall,
+        ...webData,
+    };
+};
+
+async function ScrapeWebsiteData() {
+    const tokenUrl =
+        "https://home.color.com/api/v1/get_onsite_claim?partner=natickmall";
+    // Get a request token
+    const tokenPromise = new Promise((resolve) => {
```
Comment (youssefm) [inline site-scrapers/NatickMall.js:18]: You don't actually need to re-generate a token each time. You can just hardcode any claim token you get and use it to query for availability forever. They never expire.

File: site-scrapers/NatickMall.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            })
+            .end();
+    });
+
+    const availabilityResponse = await availabilityPromise;
+    const availability = JSON.parse(availabilityResponse);
+    const results = { availability: {}, hasAvailability: false };
+    // Collect availability count by date
+    availability.reduce((memo, currentValue) => {
+        /* The availability returns and array of appointments like this:
```
Comment (youssefm) [inline site-scrapers/NatickMall.js:68]: We just pushed a change today that converts this from an array to an object wrapper with a `results` array inside it:

`{"results":[],"target_date":null}`

Sorry about the breaking change, but we don't expect any further breaking changes in the near future.

File: site-scrapers/NatickMall.js (hunk 1 above)
Comment (johnhawkinson) [reply site-scrapers/NatickMall.js:18]: How would that be provable? How do we know it and know those unwritten guarantees might not go away?

Note that this code has been merged (Feb. 19) and is in production now.

File: site-scrapers/NatickMall.js (hunk 2 above)
Comment (livgust) [reply site-scrapers/NatickMall.js:68]: Thank you!!

**Pull request merged**